Merge develop to master following B2CA-1752

agrojean-ledger opened the pull request.

3noch commented:
In the future, please open a PR against our repo. Our master has diverged from yours.

spalmer25 committed:
[.github/workflows] handle artifact using v4

spalmer25 committed:
[test] fix stax tests

ragger.firmware.stax has been replaced by ragger.firmware.touch with
the introduction of flex device

spalmer25 committed:
[test] fix stax snapshots

spalmer25 committed:
[test] ignore speculos disconnection errors when closing the app

ajinkyasraj committed:
Merge pull request #243 from trilitech/palmer@functori@bump-ragger-version

Bump ragger version

spalmer25 committed:
Merge pull request #242 from trilitech/palmer@functori@bump-github-artifact-version

Bump actions/upload-artifact and actions/download-artifact version to v4

ajinkyasraj committed:
Move stax fixed screens to parent folder. (#241)

update version and add provision for stax ragger tests in nano.

ajinkyasraj committed:
Use the guideline provided useCaseHomeAndSettings use case.

ajinkyasraj committed:
use useCaseAddressReview from ledger guidelines for stax

spalmer25 committed:
[test] remove unused code and snapshots

spalmer25 committed:
[test] use assert_home, assert_info, assert_settings to ease reading

spalmer25 committed:
[test] use functions where the code is duplicated

spalmer25 committed:
[test] move golden as argument

spalmer25 committed:
[test] explicit TezosAppScreen field

Move the ChoiceList __settings as layout

spalmer25 committed:
[test] customize use_cases for our app

spalmer25 committed:
[test] introduce a ticker pause/resume contextmanager

spalmer25 committed:
[test] remove hard-coded stax naming

spalmer25 committed:
[flex] allow flex app build

spalmer25 committed:
[test] add test for flex

spalmer25 committed:
[flex] add flex on doc and CI

ajinkyasraj committed:
Merge pull request #246 from trilitech/ajinkyaraj-23@239-use-latest-nbgl-usecase-functions-for-stax

Use the guideline provided useCaseHomeAndSettings use case.

spalmer25 committed:
[stax] fix warning

by changing `C_round_warning_64px with `C_Important_Circle_64px` that
is similar

spalmer25 committed:
[test] fix test according to the new stax display/feature

 - screen contains more data: number of screen reduced
 - swipe instead of tap
 - QR button has moved

spalmer25 committed:
[test] fix ragger touch positions

Position are fixed in Ragger v1.21.0

spalmer25 committed:
Merge pull request #249 from trilitech/palmer@functori@stax-tests-update

Stax tests update

ajinkyasraj committed:
Merge pull request #247 from trilitech/palmer@functori@flex-support

Flex support

ajinkyasraj committed:
Replace blindsigning flow to match the guidelines

If blindsigning is required, show two warning screens , the 2nd being details about the error.
Standardize the msgs shown in confirm trasaction/rejection screens using macros.

ajinkyasraj committed:
Remove blindsign setting button from settings screen on stax/flex

ajinkyasraj committed:
Update snapshots for new blindsigning flow
1. test_basic.py - remove blindsigning settings from settings menu
2. test_blindsign_too_deep.py - Remove blindsign enabled section and consolidate two warning screens in one utils function.
3. test_blindsign_too_deep_rejections.py - Address changes due to two warning screens. Note that first warning screen has reverse confirm/reject buttons
4. test_blindsign_too_large.py - Remove blindsign enabled section.
5. test_parsing_errors.py - add the blindsign warning screen. note that the warning screen has reverse confirm/reject buttons.

ajinkyasraj committed:
Make sure info.png is deleted and regenerated only in test_basic.py
info.png is deleted on unrelated golden snapshot generations.

ajinkyasraj committed:
Update blindsign msg

Use standard practice in constructoin of blindsign warning screen 2 msg. 
Update url to Tezos-ledger

ajinkyasraj committed:
Merge pull request #248 from trilitech/ajinkyaraj-23@remove-blindsign-button

Change blindsign flow according to guideline.

spalmer25 committed:
Add Ticket to the formatter

spalmer25 committed:
Merge pull request #255 from trilitech/palmer@functori@add-michelson-primitive-D_Ticket

Add michelson `D_Ticket` to the formatter

spalmer25 committed:
[blindsigning] remove blindsigning parameters

Blindsigning always activated.
i.e. if the message to sign is not parsable, alert risk and ask sign
without showing precise message's information

spalmer25 committed:
[glyphs] remove unused glyphs

spalmer25 committed:
[test] removes unused snapshots

ajinkyasraj committed:
Update tests snapshots for touch devices

ajinkyasraj committed:
Use useCaseStreamingStart to start the display stream for stax/flex display

ajinkyasraj committed:
Reset global error code at the beginning of each stream

ajinkyasraj committed:
Use useCaseReviewStreamingContinue and integrate with ui_nav_cb() function

spalmer25 committed:
[test] remove redundant code

- ledger-app-builder == 3.30.0
- ragger             == 1.21.1
- speculos           == 0.9.6

spalmer25 committed:
[touch] change the tiny-url

From `tinyurl.com/Tezos-ledger` to `bit.ly/ledger-tez`.
In order to have the same url on nano device.
Change because the old one does not fit properly on nano screens.

spalmer25 committed:
[blind-signing] Reset screens stream at blindsigning requests

spalmer25 committed:
[blindsigning] update blindsigning request flow to a standard flow

spalmer25 committed:
[format] do not clang-format on the refill error flow

To make the flow easier to read

spalmer25 committed:
[ui] allowing icon only on top

i.e. PB for our current usage

spalmer25 committed:
Merge pull request #257 from trilitech/palmer@functori@standard-nano-blindsigning

Standard nano blindsigning

ajinkyasraj committed:
Merge pull request #258 from trilitech/254-use-usecasestreaming-for-stax

254 use usecasestreaming for stax

spalmer25 committed:
[ui] center the lines on HOME mask

spalmer25 committed:
[blindsigning] use HOME mask for warning screen

spalmer25 committed:
[ux.bagls] fix bagls length

TZ_SCREEN_LINES + background + left & right arrow + optionnal icon

ajinkyasraj committed:
Remove duplicate reject UI page. 

The reviewStreamingFinish already provides a reject confirmation page.

ajinkyasraj committed:
Test removal of duplicate reject UI

spalmer25 committed:
[layout_type] associate layouts differentiated only by HOME_MASK

ajinkyasraj committed:
Merge pull request #262 from trilitech/261-duplicate-rejection-pages-at-last-signing-page

261 duplicate rejection pages at last signing page

ajinkyasraj committed:
Merge pull request #263 from trilitech/palmer@functori@fix-blindsigning-warning-arrows

Fix blindsigning warning arrows

ajinkyasraj committed:
Merge pull request #264 from trilitech/palmer@functori@bump-ledger-app-builder-version

Bump `ledger-app-builder` version

ajinkyasraj committed:
Use useCaseViewDetails to show tag-value pair which does not fit in single page. 

- fixes #260

agrojean-ledger committed:
Merge pull request #7 from trilitech/main

Flex upgrade
